db/dcrpg/pgblockchain.go

```diff
@@ -430,14 +444,18 @@ type DBInfo struct {
 // parameters. By default, duplicate row checks on insertion are enabled. See
 // NewChainDBWithCancel to enable context cancellation of running queries.
 func NewChainDB(dbi *DBInfo, params *chaincfg.Params, stakeDB *stakedb.StakeDatabase,
-	devPrefetch, hidePGConfig bool, addrCacheCap int, mp rpcutils.MempoolAddressChecker) (*ChainDB, error) {
+	devPrefetch, hidePGConfig bool, addrCacheCap int, mp rpcutils.MempoolAddressChecker,
+	dataDir string) (*ChainDB, error) {
```

Comment:
Here's another place where the architecture is specific piparser's needs.  The `dataDir` is only used to construct the piparser, but we've had to change the signature of `ChainDB`'s constructor for this.  It would be preferable to construct the parser outside of `ChainDB` (i.e. main/rebuilddb2), pass it in through a `ProposalFetcher` argument.  This approach applies regardless of how the `ProposalFetcher` is implemented.